aws/sra-test/integration-tests/aws-sdk-s3/tests/sra_test.rs

```diff
@@ -3,282 +3,75 @@
-};
-use aws_sdk_s3::primitives::SdkBody;
+
+use aws_sdk_s3::Client;
+
+use aws_smithy_client::dvr;
+use aws_smithy_client::dvr::MediaType;
 use aws_smithy_client::erase::DynConnector;
-use aws_smithy_client::test_connection::TestConnection;
-use aws_smithy_http::endpoint::SharedEndpointResolver;
-use aws_smithy_runtime::client::connections::adapter::DynConnectorAdapter;
-use aws_smithy_runtime::client::orchestrator::endpoints::DefaultEndpointResolver;
-use aws_smithy_runtime_api::client::interceptors::error::ContextAttachedError;
-use aws_smithy_runtime_api::client::interceptors::{Interceptor, InterceptorContext, Interceptors};
-use aws_smithy_runtime_api::client::orchestrator::{
-    BoxError, ConfigBagAccessors, Connection, HttpRequest, HttpResponse, RequestTime, TraceProbe,
-};
-use aws_smithy_runtime_api::client::retries::RetryClassifiers;
+
+use aws_smithy_runtime_api::client::orchestrator::{ConfigBagAccessors, RequestTime};
 use aws_smithy_runtime_api::client::runtime_plugin::RuntimePlugin;
 use aws_smithy_runtime_api::config_bag::ConfigBag;
-use aws_smithy_runtime_api::type_erasure::TypedBox;
-use aws_types::region::SigningRegion;
-use aws_types::SigningService;
-use http::Uri;
-use std::sync::Arc;
-use std::time::{Duration, UNIX_EPOCH};
 
-// TODO(orchestrator-test): unignore
-#[ignore]
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+const LIST_BUCKETS_PATH: &str = "test-data/list-objects-v2.json";
+
 #[tokio::test]
 async fn sra_test() {
     tracing_subscriber::fmt::init();
 
-    let conn = TestConnection::new(vec![(
-        http::Request::builder()
-            .header("authorization", "AWS4-HMAC-SHA256 Credential=ANOTREAL/20210618/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date;x-amz-security-token;x-amz-user-agent, Signature=ae78f74d26b6b0c3a403d9e8cc7ec3829d6264a2b33db672bf2b151bbb901786")
-            .uri("https://test-bucket.s3.us-east-1.amazonaws.com/?list-type=2&prefix=prefix~")
-            .body(SdkBody::empty())
-            .unwrap(),
-        http::Response::builder().status(200).body("").unwrap(),
-    )]);
+    let conn = dvr::ReplayingConnection::from_file(LIST_BUCKETS_PATH).unwrap();
 
     let config = aws_sdk_s3::Config::builder()
         .credentials_provider(Credentials::for_tests())
         .region(Region::new("us-east-1"))
-        .http_connector(conn.clone())
+        .http_connector(DynConnector::new(conn.clone()))
         .build();
-    let client = aws_sdk_s3::Client::from_conf(config);
+    let client = Client::from_conf(config);
+    let fixup = FixupPlugin {
+        client: client.clone(),
+        timestamp: UNIX_EPOCH + Duration::from_secs(1624036048),
+    };
 
-    let _ = dbg!(
+    let resp = dbg!(
         client
             .list_objects_v2()
             .config_override(aws_sdk_s3::Config::builder().force_path_style(false))
             .bucket("test-bucket")
             .prefix("prefix~")
-            .send_v2()
+            .send_v2(Some(fixup))
```

Comment:
I could see us adding a version of the send method like this that's only compiled for tests. I don't like the idea of users having to specify `None` for most requests though.

```rust
#[cfg(test)]
send_with_plugin(plugin: RuntimePlugin) -> Output { ... }
```